### ISSUE TITLE
Wildcards in routes?

### ISSUE THREAD

**Issue by sebastienratez** [juliomatcom/one-file-php-framework#6]
Hi there!

I am trying to implement a wildcard in the routes, such as showing in https://auth0.com/docs/quickstart/backend/php (see `$router->before('GET', '/secured/.*', function() {...`)

Do you think there would be something similar in one-php? Basically what that code is doing is adding a controller before any other to make sure any route in this path requires authentication.

Thanks

**Comment by juliomatcom** [comment]
Hello! 
Cool, would be nice, any pull request is welcome 🤗

**Comment by sebastienratez** [comment]
That's a good idea. I have a long week-end coming up so I might have time
to at least evaluate the work to be done

Cheers to that,
S.

**Comment by sebastienratez** [comment]
In fact it looks very easy. Question: why do you decided to exit traverseRoutes after the first matching route is found and executed? Would there be a problem to execute all the matching routes?
Other than this, I suppose the idea is to update matchUriWithRoute and CompareSegments. That shouldn't take too long, I'll see if I can do it on thursday or friday

**Comment by juliomatcom** [comment]
> Would there be a problem to execute all the matching routes?

No, as long as we return the proper `boolean` at the end of the loop. 

Maybe we can add the `before` as an special method and execute it always first :thinking: .. thinking we need to modify the `processUri` regex to match `*` routes too.

**Comment by sebastienratez** [comment]
before would be a good idea, but logically it would mean that for any
matching route, that before callback would get executed. The wildcard's
purpose is complementary: for example you could have a call to
`->before("/context/.*", callback);` as well. Maybe as a second feature
request?
